File: user-service/src/main/java/rs/raf/user_service/domain/mapper/PermissionMapper.java
package rs.raf.user_service.domain.mapper;

import rs.raf.user_service.domain.dto.PermissionDto;
import rs.raf.user_service.domain.entity.Permission;

public class PermissionMapper {

    public static PermissionDto toDTO(Permission permission) {
        if (permission == null) {
            return null;
        }

        PermissionDto dto = new PermissionDto();
        dto.setId(permission.getId());
        dto.setName(permission.getName());
        return dto;
    }

    public static Permission toEntity(PermissionDto dto) {
        if (dto == null) {
            return null;
        }

        Permission permission = new Permission();
        permission.setId(dto.getId());
        permission.setName(dto.getName());
        return permission;
    }
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/mapper/PermissionMapper.kt
package pack.userservicekotlin.domain.mapper

import pack.userservicekotlin.domain.dto.permission.PermissionRequestDto
import pack.userservicekotlin.domain.dto.permission.PermissionResponseDto
import pack.userservicekotlin.domain.entities.Permission

fun Permission?.toDto(): PermissionResponseDto? {
    if (this == null) return null

    return PermissionResponseDto(
        id = id,
        name = name,
    )
}

fun PermissionRequestDto?.toEntity(): Permission? {
    if (this == null) return null

    val permission = Permission()
    permission.id = id
    permission.name = name
    return permission
}
